### PR TITLE
feat(): support cli hooks

### PR DESCRIPTION
**Pull request by KiritaniAyaka**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #1804 

## What is the new behavior?

Compiling hooks ts file before cli compile source code. Then import these files and running them at specific time.

The CLI will read `hooks` field of configuration file to run multiple hooks files.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

Only `beforeCompile` and `afterCompile` hooks are supported now. I have no ways to implement `beforeCompileSource` and `afterCompileSource` hooks mentioned at #1804 . Hope anyone could help me on this.

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -2,4 +2,5 @@
   "singleQuote": true,
   "trailingComma": "all",
   "quoteProps": "preserve",
+  "endOfLine": "crlf"
```
Comment (kamilmysliwiec) [inline .prettierrc:5]:
```suggestion
```

File: actions/build.action.ts [hunk 2]
```diff
@@ -134,6 +141,57 @@ export class BuildAction extends AbstractAction {
       );
     }
 
+    const hooksCompileProgram = createProgram(configuration.hooks, {
```
Comment (kamilmysliwiec) [inline actions/build.action.ts:144]: What if hooks are not specified?

**Comment by Fgc17** [comment]
Why was this closed? This feature would be very useful for appending scripts to watch mode.

**Comment by evilive3000** [comment]
I don't use the nestjs documented way of working with swagger. However, I need to compile `openapi.yaml` before running the application. At the moment I don't see a way for me to link the application startup to these side effects? I know it is possible to configure copying of non ts files with `watchAssets`, but it is not enough for me to just copy, I also need to compile from many files into one. Can you tell me how to realize it?

**Pull request closed without merging**